apps/recnet-api/src/modules/user/user.controller.ts

```diff
@@ -258,4 +260,45 @@ export class UserController {
     const { type, channels } = dto;
     return this.userService.createOrUpdateSubscription(userId, type, channels);
   }
+
+  @ApiOperation({
+    summary: "Get Slack OAuth info",
+    description: "Get the current user's Slack OAuth info.",
+  })
+  @Get("subscriptions/slack/oauth")
+  @ApiBearerAuth()
+  @Auth()
+  public async getSlackOauthInfo(
+    @User() authUser: AuthUser
+  ): Promise<GetSlackOauthInfoResponse> {
+    const { userId } = authUser;
+    return this.userService.getSlackOauthInfo(userId);
+  }
+
+  @ApiOperation({
+    summary: "Slack OAuth",
+    description: "Slack OAuth",
+  })
+  @Post("subscriptions/slack/oauth")
+  @ApiBearerAuth()
+  @Auth()
+  public async slackOauth(
+    @User() authUser: AuthUser,
+    @Body() dto: SlackOauthDto
+  ): Promise<GetSlackOauthInfoResponse> {
+    const { userId } = authUser;
+    return this.userService.installSlack(userId, dto.code);
+  }
+
+  @ApiOperation({
+    summary: "Delete Slack OAuth",
+    description: "Delete Slack OAuth",
+  })
+  @Delete("subscriptions/slack/oauth")
+  @ApiBearerAuth()
+  @Auth()
+  public async deleteSlackOauth(@User() authUser: AuthUser): Promise<void> {
+    const { userId } = authUser;
+    return this.userService.deleteSlack(userId);
+  }
```

Comment:
I think you forget to use `@UsePipes(new ZodValidationBodyPipe(SCHEMA))` for these endpoints

Reply:
Good catch